pkg/curatedpackages/packageinstaller.go

```diff
@@ -0,0 +1,86 @@
+package curatedpackages
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/utils/urls"
+)
+
+type PackageController interface {
+	InstallController(ctx context.Context) error
+}
+
+type PackageHandler interface {
+	CreatePackages(ctx context.Context, fileName string, kubeConfig string) error
+}
+
+type Installer struct {
+	packageController PackageController
+	spec              *cluster.Spec
+	packageClient     PackageHandler
+	packagesLocation  string
+}
+
+func NewInstaller(installer ChartInstaller, runner KubectlRunner, spec *cluster.Spec, packagesLocation string) *Installer {
+	pcc := newPackageController(installer, runner, spec)
+
+	pc := NewPackageClient(
+		nil,
```

Comment:
What is this nil again?

Reply:
This is an optional item for a package (I thought I responded to your comment from before). The amount of changes required will increase the changes but since it is not a major issue, I'll add it as a separate pr?